src/main/java/org/apache/sysml/runtime/instructions/cp/ListObject.java

```diff
@@ -44,30 +46,44 @@ public ListObject(List<Data> data, List<String> names) {
 		_data = data;
 		_names = names;
 	}
-	
+
+	public boolean isParameterizedList() {
+		return _names != null;
+	}
+
+	public List<Data> getData() {
+		return _data;
+	}
+
+	public long getDataSize() {
+		Optional<Long> result = _data.stream().filter(data -> data instanceof CacheableData)
+				.map(data -> ((CacheableData) data).getDataSize()).reduce((l1, l2) -> l1 + l2);
+		return result.get();
+	}
```

Comment:
1) Note that the getDataSize method only returns the in-memory data size if a matrix/frame is pinned into memory.

2) Simply the reduce.get to a plain sum